### ISSUE TITLE
Ignore editor backups in renewal hooks

### ISSUE THREAD

**Issue by lechner** [certbot/certbot#7107]
Hi,

Would it be safer to exclude editor backups (*~) when renewal-hooks are run?

Per your request, it was posted [here](https://community.letsencrypt.org/t/editor-backup-files-executed-as-renewal-hooks/94750) first.  Thank you!

Kind regards,
Felix Lechner

## My operating system is (include version):

Debian testing

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):

0.31.0-1 via apt

## I ran this command and it produced this output:

certbot renew --force-renewal

## Certbot's behavior differed from what I expected because:

Editor backup files are run when executing renewal hooks. I expected them to be ignored.

## Here is a Certbot log showing the issue (if available):

```
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Plugins selected: Authenticator nginx, Installer nginx
Renewing an existing certificate
Running deploy-hook command: /etc/letsencrypt/renewal-hooks/deploy/slapd-wallace
Running deploy-hook command: /etc/letsencrypt/renewal-hooks/deploy/slapd-wallace~
```

**Comment by bmw** [comment]
Thanks for the well written issue @lechner! I agree this is a good idea.

If someone wants to take this on, the function where this should probably be done is https://github.com/certbot/certbot/blob/926c8c198cfb6158b7ad59988702c30e33f13d53/certbot/hooks.py#L260

**Comment by lechner** [comment]
I filed #7109 but, as a contributor otherwise unfamiliar with the project's customs, I did not include an entry in `CHANGELOG.md`.

This pull request was tested on a live system, where it produced the following output even though a backup file was present:
```
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Plugins selected: Authenticator nginx, Installer nginx
Renewing an existing certificate
Running deploy-hook command: /etc/letsencrypt/renewal-hooks/deploy/slapd-wallace

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```

**Comment by bmw** [comment]
This was fixed in #7109.
